pallets/foreign-investments/src/impls.rs

```diff
@@ -132,14 +138,12 @@ impl<T: Config> ForeignInvestment<T::AccountId> for Pallet<T> {
 		who: &T::AccountId,
 		investment_id: T::InvestmentId,
 	) -> Result<T::Balance, DispatchError> {
-		Ok(T::Investment::investment(who, investment_id)?.ensure_add(
-			Swaps::<T>::pending_amount_for(
-				who,
-				investment_id,
-				Action::Investment,
-				pool_currency_of::<T>(investment_id)?,
-			),
-		)?)
+		Ok(match ForeignInvestmentInfo::<T>::get(who, investment_id) {
+			Some(info) => info
+				.invested_foreign_amount
+				.ensure_add(info.pending_increase_swap(who, investment_id)?)?,
+			None => T::Balance::default(),
+		})
```

Comment:
Nit: Could `.map(|info| ...).unwrap_or_default()`. Great that I didn't think of that when I proposed the change myself 😅

Reply:
Good point. I tried it, but the inner closure of `map` returns a result, so I could not later use `unwrap_or_default()`, and everything seemed ugly.